### PR TITLE
refactor(workflow.py): use lxml's schematron for validation

### PR DESCRIPTION
**Pull request by macintoshpie**
Addresses #49 

### Changes
- (non-breaking) use lxml's isoschematron for validating use cases. This allows us to give correct line numbers when a test fails.

### Required Updates
- The L000 use case needs to be slightly refactored. isoschematron validates the schematron file, and expects `phase` to come between `ns` and `include`. Github won't let me drop it here, so ping me and I can share the updated version if you don't want to update yourself.

### Screenshots
Examples comparing the new (left) and current (right) error reporting
Screen Shot 2020-05-19 at 3 51 57 PM
Screen Shot 2020-05-19 at 3 52 19 PM

### CODE REVIEW

File: bsyncviewer/lib/validator/workflow.py [hunk 1]
```diff
@@ -165,6 +192,9 @@ def process_use_case(self, use_case):
         return results
 
     def process_errors(self, results):
+        def format_error(error):
+            return f'line {error["line"]}: element {error["element"]}: {error["message"]}'
+
```
Comment (macintoshpie) [inline bsyncviewer/lib/validator/workflow.py:196]: Happy to change this if a different format is preferred

File: bsyncviewer/testsuitelib/energyElements.sch [hunk 2]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<schema xmlns="http://purl.oclc.org/dsdl/schematron" queryBinding="xslt2">
+<schema xmlns="http://purl.oclc.org/dsdl/schematron" queryBinding="xslt1">
```
Comment (macintoshpie) [inline bsyncviewer/testsuitelib/energyElements.sch:2]: lxml only supports xslt1 and will fail if we use xslt2 for queryBinding

File: Dockerfile [hunk 3]
```diff
@@ -43,14 +43,6 @@ WORKDIR /srv/selection-tool
 COPY /requirements.txt /srv/selection-tool/requirements.txt
 RUN pip install -r requirements.txt
 
-#install the schematron-nokogiri gem from BuildingSync org
-WORKDIR /srv
-RUN git clone https://github.com/BuildingSync/schematron
-WORKDIR /srv/schematron
-
-RUN gem build schematron-nokogiri.gemspec
-RUN gem install schematron-nokogiri-0.0.3.gem
```
Comment (macintoshpie) [inline Dockerfile:52]: Don't need this anymore. wasn't sure if we still needed to install `ruby` and `ruby-dev` above, but kept them anyways.

File: bsyncviewer/testsuitelib/energyElements.sch (hunk 2 above)
Comment (kflemin) [reply bsyncviewer/testsuitelib/energyElements.sch:2]: I can't believe that's all i would have had to change to get lxml to work!  :)

File: Dockerfile (hunk 3 above)
Comment (kflemin) [reply Dockerfile:52]: we don't need ruby and ruby-dev anymore, no.  You can remove

File: bsyncviewer/lib/validator/workflow.py (hunk 1 above)
Comment (kflemin) [reply bsyncviewer/lib/validator/workflow.py:196]: this looks good!

**Pull request merged**